ovn-tester/ovn_workload.py

```diff
@@ -115,208 +120,218 @@ def __init__(self, phys_node, container, mgmt_net, mgmt_ip,
         ext_ls_rp_name = f'ext-to-gw-{self.container}'
         lrp_ip = netaddr.IPAddress(self.ext_net.last - 1)
-        self.ext_rp = cluster.nbctl.lr_port_add(
+        self.ext_rp = await cluster.nbctl.lr_port_add(
             self.gw_router, ext_lrp_name, RandMac(), lrp_ip,
             self.ext_net.prefixlen
         )
-        self.ext_gw_rp = cluster.nbctl.ls_port_add(
+        self.ext_gw_rp = await cluster.nbctl.ls_port_add(
             self.ext_switch, ext_ls_rp_name, self.ext_rp
         )
 
         # Configure physnet.
-        self.physnet_port = cluster.nbctl.ls_port_add(
+        self.physnet_port = await cluster.nbctl.ls_port_add(
             self.ext_switch, f'provnet-{self.container}', ip="unknown"
         )
-        cluster.nbctl.ls_port_set_set_type(self.physnet_port, 'localnet')
-        cluster.nbctl.ls_port_set_set_options(
+        await cluster.nbctl.ls_port_set_set_type(self.physnet_port, 'localnet')
+        await cluster.nbctl.ls_port_set_set_options(
             self.physnet_port,
             f'network_name={cluster.brex_cfg.physical_net}'
         )
 
         # Route for traffic entering the cluster.
         rp_gw = netaddr.IPAddress(self.gw_net.last - 1)
-        cluster.nbctl.route_add(self.gw_router, cluster.net, str(rp_gw))
+        await cluster.nbctl.route_add(self.gw_router, cluster.net, str(rp_gw))
 
         # Default route to get out of cluster via physnet.
         gr_def_gw = netaddr.IPAddress(self.ext_net.last - 2)
-        cluster.nbctl.route_add(self.gw_router, gw=str(gr_def_gw))
+        await cluster.nbctl.route_add(self.gw_router, gw=str(gr_def_gw))
 
         # Force return traffic to return on the same node.
-        cluster.nbctl.run(f'set Logical_Router {self.gw_router.name} '
-                          f'options:lb_force_snat_ip={gr_gw}')
+        await cluster.nbctl.run(f'set Logical_Router {self.gw_router.name} '
+                                f'options:lb_force_snat_ip={gr_gw}')
 
         # Route for traffic that needs to exit the cluster
         # (via gw router).
-        cluster.nbctl.route_add(cluster.router, str(self.int_net),
-                                str(gr_gw), policy="src-ip")
+        await cluster.nbctl.route_add(cluster.router, str(self.int_net),
+                                      str(gr_gw), policy="src-ip")
 
         # SNAT traffic leaving the cluster.
-        cluster.nbctl.nat_add(self.gw_router, external_ip=str(gr_gw),
-                              logical_ip=cluster.net)
+        await cluster.nbctl.nat_add(self.gw_router, external_ip=str(gr_gw),
+                                    logical_ip=cluster.net)
 
     @ovn_stats.timeit
-    def provision_port(self, cluster, passive=False):
-        name = f'lp-{self.id}-{self.next_lport_index}'
-        ip = netaddr.IPAddress(self.int_net.first + self.next_lport_index + 1)
+    async def provision_port(self, cluster, passive=False):
+        async with self.lock:
```

Comment:
As mentioned in the commit log that introduces the lock, this lock is not needed.  There's no `await` in the "critical section".  I would remove it.